swarm/src/handler/one_shot.rs

```diff
@@ -213,7 +213,9 @@ where
             }
             ConnectionEvent::DialUpgradeError(DialUpgradeError { error, .. }) => {
                 if self.pending_error.is_none() {
+                    log::debug!("DialUpgradeError: {error:?}");
                     self.pending_error = Some(error);
```

Comment:
I think we should remove this, otherwise we will still close the connection later!

Reply:
Anything within `pending_error` will be returned via `ProtocolsHandlerEvent::Close` which effectively closes the connection. In this case, we don't want that but set `keep_alive` to `No` which, if not other handler needs the connection will result in closing it.